java/hiro-client/src/main/java/co/arago/hiro/client/rest/AuthenticatedRestClient.java

```diff
@@ -417,7 +417,7 @@ private static Response checkResponse(Response response) throws Exception {
     if (status >= 200 && status <= 399) {
       return response;
     } else {
-      throw new HiroException(tryUnwrap(response.getResponseBody(), response.getStatusText()), status);
+      throw new HiroException(tryUnwrap(response.getResponseBody(), response.getStatusText()), status, JSONValue.parse(response.getResponseBody()));
```

Comment:
the "tryUnwrap" does already get the upstream error
if available.  the new solution will fail if response is non-JSON which
could happen in some error scenarios